SimTracker/TrackerHitAssociation/test/TestAssociator.cc

```diff
@@ -109,7 +109,7 @@ void TestAssociator::printRechitSimhit(const edm::Handle<edmNew::DetSetVector<re
           edm::LogVerbatim("TrackAssociator")
               << " simtrack ID = " << m.trackId() << "                            Simhit Pos = " << m.localPosition();
           // Seek the smallest residual
-          if (const SiPixelRecHit* dummy = dynamic_cast<const SiPixelRecHit*>(&rechit)) {
+          if (dynamic_cast<const SiPixelRecHit*>(&rechit)) {
```

Comment:
@iarspider , likely problem of test is not due to this modification but I would change the code differently:

auto ptr = dynamic_cast<const SiPixelRecHit*>(&rechit));
if (nullptr != ptr) {

This place will be more clear both for non-expert of the code and for compiler, which will optimise executable.